.github/workflows/release-candidate.yml

```diff
@@ -0,0 +1,107 @@
+name: Release Candidate
+
+on:
+  # Manual Trigger GH CLI -> gh workflow run release-candidate.yml --ref <branch-name>
+  workflow_dispatch:
+
+jobs:
+  run-python-tests:
+    uses: ./.github/workflows/python-versions.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-javascript-tests:
+    uses: ./.github/workflows/js-tests.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-py-prod-deps-smoke-test:
+    uses: ./.github/workflows/py-prod-deps-smoke-test.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  run-cypress-tests:
+    uses: ./.github/workflows/cypress.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  build-release-candidate:
+    runs-on: ubuntu-latest
+
+    needs: [run-python-tests, run-javascript-tests, run-py-prod-deps-smoke-test, run-cypress-tests]
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Set Environment Variables
+        uses: ./.github/actions/branch
+        with:
+          branch: ${{ github.ref_name }}
+      - name: Set desired version from branch name
+        env:
+          BRANCH: ${{ env.BRANCH }}
+        run: |
+          echo "DESIRED_VERSION=$(echo "${BRANCH}" | tr -d "release/")" >> $GITHUB_ENV
+      - name: Set final versions for pre-release and update version
+        env: 
+          DESIRED_VERSION: ${{ env.DESIRED_VERSION }}
+        run: |
+          # Start at the latest existing RC for this version if one exists,
+          # so the following update will increment to the right RC version
+          LATEST_MATCHING_RC=$(curl https://pypi.org/pypi/streamlit/json | jq ".releases | keys | map(select(contains(\"rc\"))) | map(select(contains(\"$DESIRED_VERSION\"))) | sort | .[-1]" | tr --delete '"')
+          if [ "$LATEST_MATCHING_RC" != "null" ]; then python scripts/update_version.py "$LATEST_MATCHING_RC"; fi
+
+          STREAMLIT_RELEASE_SEMVER=$(python scripts/get_prerelease_version.py "$DESIRED_VERSION")
+          echo "STREAMLIT_RELEASE_VERSION=$(echo "$STREAMLIT_RELEASE_SEMVER" | sed s/\-rc\./rc/)" >> $GITHUB_ENV
+          python scripts/update_version.py "$STREAMLIT_RELEASE_SEMVER"
+      - name: Create Package
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          make package
+      - name: Run CLI regression tests
+        run: make cli-regression-tests
```

Comment:
nit: Should this run as its own job like the other tests are?

Reply:
My guess is that this was done originally because running the CLI regression tests needs the `STREAMLIT_RELEASE_VERSION` variable created a couple steps above